Assets/PlayroomKit/PlayroomKit.cs

```diff
@@ -222,25 +226,32 @@ private static void OnPlayerJoinWrapperCallback(string id)
             {
                 callback?.Invoke(player);
             }
-            // onPlayerJoinCallback?.Invoke(player);
         }
 
-
-
-
-        public static void OnPlayerJoin(Action<Player> onPlayerJoinCallback)
+        public static Action OnPlayerJoin(Action<Player> onPlayerJoinCallback)
         {
             if (!isPlayRoomInitialized)
             {
                 Debug.LogError("PlayroomKit is not loaded!. Please make sure to call InsertCoin first.");
+                return null;
             }
             else
             {
                 if (IsRunningInBrowser())
                 {
-                    // onPlayerJoinCallback = playerCallback;
-                    OnPlayerJoinCallbacks.Add(onPlayerJoinCallback);
-                    OnPlayerJoinInternal(__OnPlayerJoinCallbackHandler);
+                    if (!OnPlayerJoinCallbacks.Contains(onPlayerJoinCallback))
+                    {
+                        OnPlayerJoinCallbacks.Add(onPlayerJoinCallback);
+                    }
+                    var CallbackID = OnPlayerJoinInternal(__OnPlayerJoinCallbackHandler);
+
+                    void Unsubscribe()
+                    {
+                        OnPlayerJoinCallbacks.Remove(onPlayerJoinCallback);
+                        UnsubscribeOnPlayerJoin(CallbackID);
```

Comment:
Is there a specific reason why `OnPlayerJoinCallbacks.Remove(onPlayerJoinCallback);` and `UnsubscribeOnPlayerJoin(CallbackID);` are in different lines? (just need some reasoning)

Reply:
No specific reason. I mean I can move the remove thing to `UnsubscribeOnPlayerJoin()` function, but that won't change anything will still need to pass `onPlayerJoinCallback` as an argument.

Reply:
> No specific reason. I mean I can move the remove thing to `UnsubscribeOnPlayerJoin()` function, but that won't change anything will still need to pass `onPlayerJoinCallback` as an argument.

Whatever helps in separation of concern, would make the most sense here. I think it's alright if they are two distinct operations with no relationship to one another.